distributed/deploy/spec.py

```diff
@@ -467,10 +471,14 @@ async def _close(self):
         await super()._close()
 
     async def __aenter__(self):
-        await self
-        await self._correct_state()
-        assert self.status == Status.running
-        return self
+        try:
+            await self
```

Comment:
`await self` and `self._correct_state()` should be updated to both call `await self.close()` on Exception then you won't need the code here

Reply:
To my understanding `_correct_state` is used for all enlarge/shrink operations. So if you e.g. fail to increase the cluster size from 100 to 200 workers because you only have 150 hosts available, it should not shut down the healthy 100 workers.

Reply:
In that case the try/await self lines should be swapped?

Reply:
Ok that's a bug, can keep the await self in the try for now and we can try fixing exceptions in await self not closing later